### PR TITLE
Manage security settings passkeys

### PR DESCRIPTION
**Pull request by yyorkchan**
Demo: 

Cherry-pick on [#15](https://github.com/cychiuae/authgear-server/pull/15/commits/94b249364781eb359fd4549bf2800311b08d7b2a)

https://github.com/user-attachments/assets/389924ad-36d9-412d-984c-16848a94d8fe

### CODE REVIEW

**Comment by cychiuae** [comment]
@yyorkchan Please rebase

File: authui/src/authflowv2/components/settings-dialog.css [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@layer components {
+  .settings-dialog {
+    /* When Mobile View */
+    @apply p-8;
+
+    /* When Not Mobile View */
+    @media (min-width: theme("screens.tablet")) {
```
Comment (cychiuae) [inline authui/src/authflowv2/components/settings-dialog.css:7]: Use `@screen`

https://tailwindcss.com/docs/functions-and-directives#screen

File: pkg/lib/accountmanagement/service.go [hunk 2]
```diff
@@ -290,3 +320,90 @@ func (s *Service) ChangePassword(input *ChangePasswordInput) (*ChangePasswordOut
...
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &AddPasskeyOutput{}, nil
+}
```
Comment (cychiuae) [inline pkg/lib/accountmanagement/service.go:377]: Please list out all the possible flows of nodes and edges on `interaction` and check carefully what side effects we need to perform when adding a new identity

File: pkg/lib/accountmanagement/service.go [hunk 3]
```diff
@@ -290,3 +320,90 @@ func (s *Service) ChangePassword(input *ChangePasswordInput) (*ChangePasswordOut
...
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &RemovePasskeyOutput{}, nil
+}
```
Comment (cychiuae) [inline pkg/lib/accountmanagement/service.go:409]: Please list out all the possible flows of nodes and edges on `interaction` and check carefully what side effects we need to perform when removing a new identity

File: authui/src/authflowv2/dialog.ts [hunk 4]
```diff
@@ -59,25 +59,27 @@ function dispatchDialogCloseEnd(dialogID: string) {
...
+      // close event is TS event that targets this dialog
+      this.element.classList.remove("open");
     }
-    if (this.element.id !== e.detail.id) {
-      // close event targets other dialog
-      return;
+    // @ts-expect-error: The params property only exists on Stimulus events.  */
+    if (e instanceof PointerEvent && this.element.id === e.params.id) {
+      // close event is Stimulus event that targets this dialog
+      this.element.classList.remove("open");
```
Comment (cychiuae) [inline authui/src/authflowv2/dialog.ts:81]: Why these changes?

Comment (cychiuae) [reply authui/src/authflowv2/dialog.ts:81]: What the purposes of these?

File: authui/src/authflowv2/settingsDialog.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Controller } from "@hotwired/stimulus";
+import { dispatchDialogOpen } from "./dialog";
+
+// Assume globally only have ONE single dialog
+const DIALOG_ID = "settings-dialog";
+/**
+ * Dispatch a custom event to set captcha dialog open
+ */
```
Comment (cychiuae) [inline authui/src/authflowv2/settingsDialog.ts:8]: We shouldn't have this assumption

File: resources/authgear/templates/zh-HK/translation.json [hunk 6]
```diff
@@ -1113,6 +1115,9 @@
   "v2.page.select-account.default.title": "登入 {AppOrClientName}",
   "v2.page.select-account.default.use-another-account": "使用另一個帳戶",
   "v2.page.settings-change-password.default.navbar-title": "更改密碼",
+  "v2.page.settings-passkey.default.button-label-add-passkey": "+ 新增通行密匙",
+  "v2.page.settings-passkey.default.item-description": "{name}<br>新增時間：<span data-date=\"{rfc3339}\">{time, datetime, short} UTC</span>",
```
Comment (cychiuae) [inline resources/authgear/templates/zh-HK/translation.json:1119]: Write better markup. Don't put these markup in translation

File: resources/authgear/templates/zh-HK/translation.json [hunk 7]
```diff
@@ -1113,6 +1115,9 @@
   "v2.page.select-account.default.title": "登入 {AppOrClientName}",
   "v2.page.select-account.default.use-another-account": "使用另一個帳戶",
   "v2.page.settings-change-password.default.navbar-title": "更改密碼",
+  "v2.page.settings-passkey.default.button-label-add-passkey": "+ 新增通行密匙",
```
Comment (cychiuae) [inline resources/authgear/templates/zh-HK/translation.json:1118]: v2.page.settings-passkey.default.add-passkey-button-label

File: authui/src/authflowv2/components/remove-button.css [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+  :root.dark {
+    --remove-btn__bg-color--disabled: var(--color-neutral-400);
+    --remove-btn__text-color--disabled: var(--color-neutral-100);
+
+    --remove-btn__bg-color--hover: var(--color-error-tertiary);
+    --remove-btn__bg-color--active: var(--color-error-tertiary);
+  }
+
+  .remove-btn {
```
Comment (cychiuae) [inline authui/src/authflowv2/components/remove-button.css:43]: Rename to `desctructive-btn`

File: authui/src/authflowv2/base.css [hunk 9]
```diff
@@ -31,7 +31,9 @@
   /* Colors */
   :root {
     --color-success: #11c624;
+    --color-error-dark: #c4003d;
     --color-error: #ff295b;
+    --color-error-tertiary: #c4003d; /* This is not correct. */
```
Comment (cychiuae) [inline authui/src/authflowv2/base.css:36]: Dont write incorrect code

Comment (cychiuae) [reply authui/src/authflowv2/base.css:36]: https://www.figma.com/design/LRE8EfCaTfYxC0lk5AeMmE/New-Auth-UI?node-id=716-21630&t=k4froDvyqJTH93wE-4

**Review by cychiuae — CHANGES_REQUESTED** [review]

File: authui/src/authflowv2/dialog.ts (hunk 4 above)
Comment (yyorkchan) [reply authui/src/authflowv2/dialog.ts:81]: Please note that the settings dialog is cherry-pick from #15

Comment (Kuixz) [reply authui/src/authflowv2/dialog.ts:81]: These are to be able to trigger dialog open/close events via Stimulus `data-action"click->dialog#open"` 
 and `data-action"click->dialog#close"` attribute, one instance of this appears in `__settings_dialog.html`.

File: resources/authgear/templates/en/web/authflowv2/__settings_dialog.html [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    (dict
+      "Type" "Cancel"
+      "Label" "Cancel"
+    )
+  )
+)
+-->
+
+{{ define "authflowv2/__settings_dialog.html" }}
+  {{ $settingsDialogID := "settings-dialog" }}
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__settings_dialog.html:21]: Pass from outside

File: resources/authgear/templates/en/web/authflowv2/__settings_dialog.html [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+          "DialogID" $settingsDialogID
+        )
+      )
+    )
+  }}
+{{ end }}
+
+{{ define "authflowv2/__settings_dialog_content.html" }}
+<form method="post" action="{{ $.FormAction }}">
+  {{ $.Ctx.CSRFField }}
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__settings_dialog.html:44]: Include this in FormContent instead

File: resources/authgear/templates/en/web/authflowv2/__settings_dialog.html [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+  <div class="flex flex-col gap-y-8 flex-1-0-auto">
+    <header class="dialog-title-description">
+      <h1 class="dialog-title">
+        {{ $.Title }}
+      </h1>
+      <h2 class="dialog-description">
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__settings_dialog.html:50]: should be p / div tag

File: resources/authgear/templates/en/web/authflowv2/__settings_dialog.html [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+      )
+    )
+  }}
+{{ end }}
+
+{{ define "authflowv2/__settings_dialog_content.html" }}
+<form method="post" action="{{ $.FormAction }}">
+  {{ $.Ctx.CSRFField }}
+  <div class="flex flex-col gap-y-8 flex-1-0-auto">
+    <header class="dialog-title-description">
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__settings_dialog.html:46]: maybe use div?

File: resources/authgear/templates/en/web/authflowv2/__settings_dialog.html [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+      <h1 class="dialog-title">
+        {{ $.Title }}
+      </h1>
+      <h2 class="dialog-description">
+        {{ $.Description }}
+      </h2>
+  </header>
+    <span class="flex flex-col gap-y-4">
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__settings_dialog.html:54]: Use div

**Comment by yyorkchan** [comment]
Closed for new PR
